### PR TITLE
[DX-1062] Add API and SDK changelog

### PR DESCRIPTION
**Pull request by sprngbk**
## Description
This PR:
* Adds `API changelog` as a dropdown for the API navbar header
image

* Adds an API changelog Markdown page as a new page in docs
image

* Adds `Changelog` as a sidebar attribute for Typescript Core SDK docs
image

## Resolved issues
Jira: https://immutable.atlassian.net/browse/DX-1062

### Before submitting the PR, please consider the following:
- [x] It's beneficial if your pull request references an issue used to discuss the pull request ahead of time. If you haven't previously created an issue, please create one and discuss your contribution with the maintainers.
- [x] Prefix your PR title with `feat: `, `fix: `, `chore: `, `docs:`, or `refactor:`.
- [x] The description should clearly illustrate what problems the pull request solves.
- [x] Ensure that the commit messages follow our guidelines.
- [x] Resolve merge conflicts (if any).
- [x] Make sure that the current branch is up to date with the `main` branch.

### For internal Immutable developers:
- [ ] If you are adding or updating documentation for an SDK, please make sure that you meet the requirements in [this doc](https://immutable.atlassian.net/wiki/spaces/PPS/pages/1916994017/SDK+documentation+guide).
- [ ] Please obtain PR approval from all of the following:
    - Your tech lead (Fallback: Team lead or engineering manager)
    - Your product manager (Fallback: Group PM)
    - Another software engineer on your team

### CODE REVIEW

**Review by ckailash — APPROVED** [review]
LFG 🚀 :shipit:

File: src/pages/api-changelog.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+id: "api-changelog"
+title: "API changelog"
+---
+
+# API changelog
```
Comment (ckailash) [inline src/pages/api-changelog.md:6]: Could we add a sidebar? Ensure that this page's design matches the rest of the docs.

**Review by ckailash — CHANGES_REQUESTED** [review]

File: docusaurus.config.js [hunk 2]
```diff
@@ -134,6 +134,13 @@ const configuration = {
             to: '/reference',
             position: 'left',
             label: 'APIs',
+            type: 'dropdown',
```
Comment (ckailash) [inline docusaurus.config.js:137]: The dropdown's design is a bit weird. Maybe some CSS changes are also required?

File: src/pages/api-changelog.md (hunk 1 above)
Comment (sprngbk) [reply src/pages/api-changelog.md]: I've added it as a page in our docs so the sidebar issue is resolved 👍

File: docusaurus.config.js (hunk 2 above)
Comment (sprngbk) [reply docusaurus.config.js:137]: Yup definitely, am on it 👍

**Review by matthewmuscat — APPROVED** [review]
LGTM!

**Pull request closed without merging**